megamek/src/megamek/client/ui/swing/boardview/TWBoardViewTooltip.java

```diff
@@ -0,0 +1,460 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.swing.boardview;
+
+import megamek.client.ui.Messages;
+import megamek.client.ui.swing.ClientGUI;
+import megamek.client.ui.swing.GUIPreferences;
+import megamek.client.ui.swing.tooltip.HexTooltip;
+import megamek.client.ui.swing.tooltip.PilotToolTip;
+import megamek.client.ui.swing.tooltip.UnitToolTip;
+import megamek.client.ui.swing.util.KeyCommandBind;
+import megamek.client.ui.swing.util.UIUtil;
+import megamek.common.*;
+import megamek.common.actions.ArtilleryAttackAction;
+import megamek.common.annotations.Nullable;
+import megamek.common.options.OptionsConstants;
+
+import java.awt.*;
+import java.util.*;
+import java.util.List;
+
+import static megamek.client.ui.swing.util.UIUtil.guiScaledFontHTML;
+import static megamek.client.ui.swing.util.UIUtil.uiWhite;
+
+public class TWBoardViewTooltip implements BoardViewTooltipProvider {
+
+    private final GUIPreferences GUIP = GUIPreferences.getInstance();
+
+    private final ClientGUI clientGui;
+    private final Game game;
+    private final BoardView bv;
+
+    public TWBoardViewTooltip(Game game, @Nullable ClientGUI clientGui, BoardView boardView) {
+        this.clientGui = clientGui;
+        this.game = game;
+        this.bv = boardView;
+    }
+
+    @Override
+    public String getTooltip(Point point, Coords movementTarget) {
+        final Coords coords = bv.getCoordsAt(point);
+        if (!game.getBoard().contains(coords)) {
+            return null;
+        }
+        Entity selectedEntity = (clientGui != null) ? clientGui.getSelectedUnit() : null;
+        Player localPlayer = localPlayer();
+        Hex mhex = game.getBoard().getHex(coords);
+
+        String result = "";
+
+        // Hex Terrain
+        if (GUIP.getShowMapHexPopup() && (mhex != null)) {
+            StringBuffer sbTerrain = new StringBuffer();
+            appendTerrainTooltip(sbTerrain, mhex, GUIP);
+            String sTrerain = sbTerrain.toString();
```

Comment:
Typo?  'sTrerain' -> 'sTerrain'?

Reply:
yes indeed. Not my typo :) I corrected it